redis/asyncio/client.py

```diff
@@ -1386,17 +1430,10 @@ async def execute(self, raise_on_error: bool = True):
 
         try:
             return await asyncio.shield(
-                conn.retry.call_with_retry(
-                    lambda: execute(conn, stack, raise_on_error),
-                    lambda error: self._disconnect_raise_reset(conn, error),
-                )
+                self._try_execute(conn, execute, stack, raise_on_error)
             )
-        except asyncio.CancelledError:
-            # not supposed to be possible, yet here we are
-            await conn.disconnect(nowait=True)
-            raise
-        finally:
-            await self.reset()
+        except RuntimeError:
```

Comment:
What happens if we Canceled (so this block raises CanceledError) and then get a RuntimeError inside the shield (unhandled by this try...except since we already exited this block)

Reply:
So the issue was that I couldn't successfully get that to descend in 100% of cases. The Cancelled isn't triggered - which looks very similar to the aiohttp bug - it should be.

But - thanks to testing in multiple python versions, I can also see that there are instances where the RuntimeError doesn't triggers, so a finally wrapper is also needed. Fascinated - I'll probably update a couple of existing bugs in the ecosystem.